api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -474,24 +473,35 @@ func (c *imageBuilder) createKanikoJobSpec(project mlp.Project, model *models.Mo
 		}
 	}
 
-	resourceRequirements := RequestLimitResources{
-		Request: Resource{
-			CPU: resource.MustParse(
-				c.config.Resources.Requests.CPU,
-			),
-			Memory: resource.MustParse(
-				c.config.Resources.Requests.Memory,
-			),
-		},
-		Limit: Resource{
-			CPU: resource.MustParse(
-				c.config.Resources.Limits.CPU,
-			),
-			Memory: resource.MustParse(
-				c.config.Resources.Limits.Memory,
-			),
-		},
-	}
+	if resource != nil {
+		// Get resourceRequirements from version endpoint configuration
+		resourceRequirements := RequestLimitResources{
+			Requests: Resource{
+				CPU:    resource.MustParse(resource.CPURequest),
+				Memory: resource.MustParse(resource.MemoryRequest),
```

Comment:
what if user only wish to overwrite one of them? like providing mem override only

@ariefrahmansyah , this should be supported right?